Ansible/roles/marvin/templates/smoketests.sh.j2

```diff
@@ -35,32 +35,92 @@ rm -f /usr/lib/python2.7/site-packages/marvin/config/*pyc
+    cloudmonkey set output json
+    # TODO: N/A
+
+    # Destroy alerts
+    for id in $(cloudmonkey list alerts listall=true keyword=vm filter=id | jq -r '.alert[] .id'); do
+        cloudmonkey delete alerts ids=$id || true
+    done
+
+    # Destroy templates, forced, since some VMs might exist.
+    for templateid in $(cloudmonkey list templates templatefilter=self filter=id | jq -r '.template[] .id'); do
+        cloudmonkey delete template id=$templateid forced=true  || true
     done
-    # Destroy routers
-    for rid in $(cloudmonkey list routers listall=true filter=id | grep '^id' | awk '{print $3}'); do
-        cloudmonkey destroy router id=$rid || true
+
+    # Destroy vms, all in parallel
+    for vmid in $(cloudmonkey list virtualmachines listall=true filter=id | jq -r '.virtualmachine[] .id'); do
+        cloudmonkey destroy virtualmachine id=$vmid expunge=true || true
     done
-    # Destroy networks
-    for netid in $(cloudmonkey list networks listall=true filter=id | grep '^id' | awk '{print $3}'); do
+
+    # Not needed to destroy VRs explicitely, will be destroyed with network/VPC
+
+    # Destroy networks only when no more VMS (otherwise will fail if network contains a VM)
+         NoVMs=$(cloudmonkey list virtualmachines filter=id  | grep id | wc -l)
+         while [ $NoVMs -ne 0 ]
+         do
+          sleep 5
+          #echo "sleeping 5 sec"
+             if [[ "$NoVMs" == '0' ]]; then
+             break
+             fi
+          NoVMs=$(cloudmonkey list virtualmachines filter=id  | grep id | wc -l)
+         done
+    for netid in $(cloudmonkey list networks listall=true filter=id | jq -r '.network[] .id'); do
         cloudmonkey delete network id=$netid || true
     done
-    # Destroy vpcs
-    for vid in $(cloudmonkey list vpcs listall=true filter=id | grep '^id' | awk '{print $3}'); do
+
+    # Destroy vpcs, only if no networks present (otherwise will fail if VPC has networks)
+         NoNets=$(cloudmonkey list networks filter=id | grep id | wc -l)
+         while [ $NoNets -ne 0 ]
+         do
+          sleep 5
+          #echo "sleeping 5 sec"
+             if [[ "$NoNets" == '0' ]]; then
+             break
+             fi
+          NoNets=$(cloudmonkey list networks filter=id | grep id | wc -l)
+         done
+    for vid in $(cloudmonkey list vpcs listall=true filter=id | jq -r '.vpc[] .id'); do
         cloudmonkey delete vpc id=$vid || true
     done
-    # Destroy alerts
-    for id in $(cloudmonkey list alerts listall=true keyword=auth-error filter=id | grep '^id' | awk '{print $3}'); do
-        cloudmonkey delete alerts ids=$id || true
+
+    # move to blocking asyn calls (just volume deletion, should be very fast), so later we can delete accounts and domains gracefully
+    cloudmonkey set asyncblock true
+
+    # Destroy data volumes when zero attached (all VMs deleted)
+      NoAttachedVolumes=$(cloudmonkey list volumes type=data  | grep virtualmachineid | wc -l)
```

Comment:
is some mechanism required to break out of this loop if a disk expunge fails, otherwise this could get stuck in an infinite loop.